### DISCUSSION TITLE
Determine if a job is currently running on the server where the runner is installed?

### DISCUSSION THREAD

**Discussion by retzero** [actions/runner#1658] [Q&A] [answered]
Hi,

I want to check whether there is a job currently in progress on the VM where runner application is installed without using Rest API.

I got below process if I `ps` with `Runner` string. 
Is it safe to judge whether a job is in progress with this process name?
What is the best way to do this?

```
root      359124   18395 30 05:34 ?        00:00:03 /home/github/actions-runner/bin/Runner.Worker spawnclient 99 105
```

**Comment by thboop** [comment, marked as answer]
The Worker client is only spawned once a job is received, and exits once the job completes. I can say this will currently tell you if a job is running, but it really isn't our supported way to do this. 

We have some backlog issues to [add better telemetry/hooks](https://github.com/actions/runner/issues/699) to what the runner is doing that we hope to do in the future.
